src/gsi/setupdbz.f90

```diff
@@ -360,97 +616,604 @@ subroutine setupdbz(obsLL,odiagLL,lunin,mype,bwork,awork,nele,nobs,is,radardbz_d
+                    Zer = 10E-8_r_kind
+                 endif
+    
+!                snow
+                 Zeso  = Cs_dry * (rho * qsexp)**(Ps_dry)
+                 if (qsexp > 10E-7_r_kind) then 
+                    if ( T1D > 274.15_r_kind .and. iqrges > 1.E-12_r_kind) then
+                       Zes = (1.47E+05_r_kind)*(qsexp*1000._r_kind)**2.67_r_kind
+                    else
+                       ! calc coeffs for dry snow based on TM MP code 
+                       call calc_coeffs_dry_snow_tm(T1D,a_dry_snow_tm,b_dry_snow_tm)
+                       smoz = a_dry_snow_tm * (qsexp*rho*oams)**b_dry_snow_tm
+                       Zes  = alpha_const_tm_dry_snow*smoz
+                    end if
+                 else
+                    Zes = 10E-8_r_kind
+                 endif
+    
+!                graupel/hail
+                 Zego  = Cg_dry * (rho * qgexp)**(Pg_dry)
+                 if ( qgexp > 10E-7_r_kind ) then
+                    if ( T1D > 274.15_r_kind .and. iqrges > 1.E-12_r_kind) then
+                       Zeg = Cg_wet*rho**1.75_r_kind*qgexp**Pg_wet
+                    else
+                       Zeg = Cg_dry*rho**1.75_r_kind*qgexp**Pg_wet
+                    end if
+                 else
+                    Zeg=10E-8_r_kind
+                 endif
+    
+                 Zeo=Zero1+Zeso+Zego
+    
+                 if(Zeo <1.0_r_kind) then
+                    Zeo=Zeo + 1.0_r_kind
+                 end if
+    
+                 Zo = ten * log10(Zeo)
+                 Zro = ten * log10(Zero1)
+                 Zso = ten * log10(Zeso)
+                 Zgo = ten * log10(Zego)
+    
+                 Ze=Zer+Zes+Zeg
+    
+!                Zelim treatment
+                 if(Ze <1.0_r_kind) then
+                    Ze=Ze + 1.0_r_kind
+                 end if
+    
+!                Convert to simulated radar reflectivity in units of dBZ
+                 Ze_orig = Ze
+                 rdBZ = ten * log10(Ze)
+                 rdBZr = ten * log10(Zer)
+                 rdBZs = ten * log10(Zes)
+                 rdBZg = ten * log10(Zeg)
+    
+!                find dqr/ddBZ, dqs/ddBZ, dqg/ddBZ (used in inner loop routine)
+!                Jacobian used for TLM and ADM
+                 if ( l_use_cvpqx ) then
+!                   rain
+                    if ( qrexp > 10E-7_r_kind .and. qscalar(P_nr) > 20._r_kind ) then
+                       jqr_num = (2*720*rho**2_r_kind*qrexp**(2_r_kind-cvpqx_pval)/&
+                                 (pi**2_r_kind*rhor**2_r_kind*qscalar(P_nr)))*ten**12_r_kind*ten**6_r_kind
+                       if (l_cvpnr) then
+                          jqnr_num = (-720*rho**2_r_kind*qrexp**2_r_kind/&
+                                     (pi**2_r_kind*rhor**2_r_kind*qscalar(P_nr)**(1_r_kind+cvpnr_pval)))&
+                                     *ten**12_r_kind*ten**6_r_kind
+                       else
+                          jqnr_num = (-720*rho**2_r_kind*qrexp**2_r_kind/&
```

Comment:
Please add _r_kind to 2 and 720
```
(2_r_kind*720_r_kind
-720_r_kind
```

Reply:
We changed the code as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-7a9c44619cb5bc5c1e48fcc0d434a8be4257007f9f4a2c8a9cd9090ff3873135R1040-R1047